### PR TITLE
Tidy up for adding engine

### PR DESCRIPTION
**Pull request by jamessizeland**

### CODE REVIEW

**Comment by ScottGibb** [comment]
Much wow, much like

File: src-tauri/src/chess/api.rs [hunk 1]
```diff
@@ -172,24 +151,24 @@ pub fn click_square(
                 // 4. if we have clicked a valid move of selected, do move
                 println!("valid move");
                 let source = selected.unwrap();
-                let mover = board[source.0][source.1].has_moved();
-                board[coord.0][coord.1] = Piece::None; // empty the destination square
-                board[source.0][source.1] = Piece::None; // take moving out of its square
-                board[coord.0][coord.1] = mover; // place moving in the new square
+                let mover = board.get(source).has_moved();
```
Comment (ScottGibb) [inline src-tauri/src/chess/api.rs:154]: I like these!

File: src-tauri/src/chess/board.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        board[6][7] = Piece::Knight(black, true);
+        board[7][7] = Piece::Rook(black, true);
+        // set up pawns
+        for col in board.iter_mut() {
+            col[1] = Piece::Pawn(white, true);
+            col[6] = Piece::Pawn(black, true);
+        }
+        BoardState(board)
+    }
+    /// Set a piece on the board
```
Comment (ScottGibb) [inline src-tauri/src/chess/board.rs:41]: Nice!

File: src-tauri/src/chess/moves.rs [hunk 3]
```diff
@@ -272,23 +267,17 @@ pub fn knight_move(sq: (usize, usize), color: Color, board: &BoardState) -> Move
 
 pub fn king_move(sq: (usize, usize), color: Color, board: &BoardState) -> MoveList {
     let mut moves: MoveList = Vec::new(); // start with empty movelist
+    #[rustfmt::skip]
```
Comment (ScottGibb) [inline src-tauri/src/chess/moves.rs:270]: Interesting! Didnt know you could do that

File: src-tauri/src/chess/pieces.rs [hunk 4]
```diff
@@ -64,15 +65,15 @@ impl Piece {
     /// Return the relative weighting value of this piece based on its type
     ///
     /// https://en.wikipedia.org/wiki/Chess_piece_relative_value
-    pub fn get_value(&self) -> isize {
+    pub fn get_value(&self) -> Option<isize> {
```
Comment (ScottGibb) [inline src-tauri/src/chess/pieces.rs:68]: I like the use of option here

File: src-tauri/src/chess/types.rs [hunk 5]
```diff
@@ -171,48 +168,18 @@ impl Display for Piece {
...
-                }
-            }
-            Piece::Rook(color, ..) => {
-                if color == Color::White {
-                    "wR"
-                } else {
-                    "bR"
-                }
-            }
+            Piece::Pawn(Color::White, ..) => "wP",
```
Comment (ScottGibb) [inline src-tauri/src/chess/types.rs:171]: What are the two dots doing here?

**Review by ScottGibb — APPROVED** [review]
very very cool!

File: src-tauri/src/chess/types.rs (hunk 5 above)
Comment (jamessizeland) [reply src-tauri/src/chess/types.rs:171]: That's short hand for "ignore any other variables I might pull out of this pattern"

So is the equivalent of 

```rust

Piece::Pawn(Colour::White, _, _) => "wP",

```

**Pull request merged**